dotcom-rendering/src/lib/fonts-css.ts

```diff
@@ -247,36 +249,31 @@ const fontList: FontDisplay[] = [
 	},
 ];
 
-const assetsUrl = (path: string): string =>
-	`https://assets.guim.co.uk/static/frontend/${path}`;
+const minifyCssString = (css: string) =>
+	css.replace(/\n/g, '').replace(/\s\s+/g, ' ');
 
-const template: (_: FontDisplay) => string = ({
-	family,
-	woff2,
-	woff,
-	ttf,
-	weight,
-	style,
-}) => `
-    @font-face {
-        font-family: "${family}";
-        src: url(${assetsUrl(woff2)}) format("woff2"),
-                url(${assetsUrl(woff)}) format("woff"),
-                url(${assetsUrl(ttf)}) format("truetype");
-        font-weight: ${weight};
-        font-style: ${style};
-        font-display: swap;
-    }
-`;
+export const getFontsCss = (offerHttp3 = false): string => {
+	let fontCss = '';
 
-const getStyleString: () => string = () => {
-	return fontList.reduce(
-		(styleString, font) => `${styleString}${template(font)}`,
-		'',
-	);
-};
+	for (const font of fontList) {
+		if (offerHttp3) {
+			font.woff2 = getHttp3Url(font.woff2);
+			font.woff = getHttp3Url(font.woff);
+			font.ttf = getHttp3Url(font.ttf);
+		}
 
-const minifyCssString = (css: string) =>
-	css.replace(/\n/g, '').replace(/\s\s+/g, ' ');
+		fontCss += `
+			@font-face {
+				font-family: "${font.family}";
+				src: url(${font.woff2}) format("woff2"),
+						url(${font.woff}) format("woff"),
+						url(${font.ttf}) format("truetype");
+				font-weight: ${font.weight};
+				font-style: ${font.style};
+				font-display: swap;
+			}
+		`;
+	}
```

Comment:
Could we avoid mutating an internal variable here with joining the array?

```ts
fontList.map((font) => { /* do thing*/ return `@font-face { /* family */ }` } ).join("\n")
```

Reply:
i don't think we need be scared of mutation _everywhere_ - it's the only object we use, its only mutated in the scope of this module, and the resulting code is explicit. 

afaict, the only reason to avoid mutating it is to avoid mutating it?

Reply:
Is `font` a reference to the font object inside the `fontList` or a new independent object?

But yeah I don't mind strongly either way.

Reply:
yeah it is mutated, i know. i think that's ok though?

Reply:
actually, no i think it's better not to mutate here, as you suggest - if `getFontsCss` is ever called more than once, invalid URLs will start being returned. removed the mutation

Reply:
For completeness, I did run a benchmark, and the mutable version is about 1.5 times faster

<img width="572" alt="image" src="https://user-images.githubusercontent.com/76776/219424679-1b1ec34d-6720-49da-9010-c74d49f2fe47.png">

```ts
import { assertEquals } from "https://deno.land/std@0.177.0/testing/asserts.ts";

const fontList = [
  { style: "bold", weight: 700 },
  { style: "regular", weight: 400 },
  { style: "light", weight: 300 },
] satisfies Array<{ style: string; weight: number }>;

const mutable = () => {
  let output = "";

  for (const { style, weight } of fontList) {
    output += `font-style: ${style}; font-weight: ${weight};`;
  }

  return output;
};
const immutable = () =>
  fontList.map(({ style, weight }) =>
    `font-style: ${style}; font-weight: ${weight};`
  ).join("");

const expected =
  "font-style: bold; font-weight: 700;font-style: regular; font-weight: 400;font-style: light; font-weight: 300;";

Deno.bench("mutable", { baseline: true }, () => {
  mutable();
});

Deno.bench("immutable", () => {
  immutable();
});

assertEquals(immutable(), expected);
assertEquals(mutable(), expected);
```